tests/ruling/utils.ts

```diff
@@ -0,0 +1,194 @@
+      rules.forEach(Rule => {
+        const rule = initRule(Rule);
+        const errorLines = runRuleOnProjectFile(rule, file, program);
+        const ruleName = (Rule as any).metadata.ruleName;
+        results = mergeErrorsToResults(
+          results,
+          ruleName,
+          getFileNameForSnapshot(file.fileName),
+          errorLines,
+        );
+      });
+    });
+  });
+  console.log("");
+  return results;
+}
+
+export function writeResults(results: Results) {
+  Object.keys(results).forEach(rule => {
+    const content: string[] = [];
+
+    const files = Object.keys(results[rule]).map(file => {
+      const lines = results[rule][file];
+      content.push(`${file}: ${lines.join()}`);
+    });
+
+    writeSnapshot(rule, content.join("\n") + "\n");
+  });
+}
+
+export function checkResults(expected: Results) {
+  const expectedRules = Object.keys(expected);
+  const actual: Results = readSnapshots(expectedRules);
+  const actualRules = Object.keys(actual);
+  let passed = true;
+
+  expectedRules.forEach(rule => {
+    const expectedFiles = expected[rule];
+    const actualFiles = actual[rule] || {};
+
+    Object.keys(expectedFiles).forEach(file => {
+      const expectedLines = expectedFiles[file];
+      const actualLines = actualFiles[file] || [];
+
+      const missingLines = lodash.difference(actualLines, expectedLines);
+      if (missingLines.length > 0) {
+        passed = false;
+        console.log("Missing issues:");
+        console.log("  * Rule:", rule);
+        console.log("  * File:", file);
+        console.log("  * Lines:", missingLines.join(", "));
+        console.log();
+      }
+
+      const extraLines = lodash.difference(expectedLines, actualLines);
+      if (extraLines.length > 0) {
+        passed = false;
+        console.log("Extra issues:");
+        console.log("  * Rule:", rule);
+        console.log("  * File:", file);
+        console.log("  * Lines:", extraLines.join(", "));
+        console.log();
+      }
+    });
+  });
+
+  return passed;
+}
+
+function initRule(Rule: any): tslint.IRule {
+  const RULE_OPTIONS: tslint.IOptions = {
+    disabledIntervals: [],
+    ruleArguments: [],
+    ruleName: "",
+    ruleSeverity: "error",
+  };
+  return new Rule(RULE_OPTIONS);
+}
+
+function getProgram(tsConfigFile: string): ts.Program {
+  return tslint.Linter.createProgram(tsConfigFile);
+}
+
+function getProgramFiles(program: ts.Program): ts.SourceFile[] {
+  return program.getSourceFiles().filter(file => !file.isDeclarationFile);
+}
+
+function runRuleOnProjectFile(rule: any, sourceFile: ts.SourceFile, program: ts.Program) {
+  let failures: tslint.RuleFailure[];
+
+  if ((rule as tslint.Rules.TypedRule).applyWithProgram) {
+    failures = rule.applyWithProgram(sourceFile, program);
+
+  } else {
+    failures = rule.apply(sourceFile);
+  }
+
+  return failures.map(failure => failure.getStartPosition().getLineAndCharacter().line + 1);
+}
+
+function mergeErrorsToResults(
```

Comment:
I would call it `addErrorsToResults`, it makes it more clear that the errors are being entered into a structure, Results, that is just a large listing of them. `Merge` makes me think that there's more into Results than just the errors